integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -479,7 +488,17 @@ object CarbonDataRDDFactory {
             segmentDetails.add(new Segment(resultOfBlock._2._1.getLoadName))
           }
         }
-        val segmentFiles = updateSegmentFiles(carbonTable, segmentDetails, updateModel.get)
+        var segmentMinMaxMap: Map[String, List[SegmentMinMax]] = Map()
+        if (!segmentMinMaxAccumulator.isZero) {
+          segmentMinMaxAccumulator.value.asScala.foreach(map => if (map.nonEmpty) {
+            segmentMinMaxMap = segmentMinMaxMap ++ map
+          })
+        }
+        val segmentFiles = updateSegmentFiles(carbonTable,
```

Comment:
Segment min max for a column means, it is a single min max value. 
Now we are storing blocklevel minmax (it is a duplicate storage as already this info is there in file footer).
we need to have a comparator here and find the lowest and highest value form all the blocks and store that value in the segment file.

Reply:
Because in the realtime, one segment can have millions of small data files. This won't improve pruning performance if we check each file.

Reply:
Handled. Please check